docs/reference/modules/scripting/security.asciidoc

```diff
@@ -1,67 +1,62 @@
 [[modules-scripting-security]]
 === Scripting and security
 
-You should never run Elasticsearch as the `root` user, as this would allow a
-script to access or do *anything* on your server, without limitations.
-
-You should not expose Elasticsearch directly to users, but instead have a
-proxy application inbetween. If you *do* intend to expose Elasticsearch
-directly to your users, then you have to decide whether you trust them enough
-to run scripts on your box or not, and apply the appropriate safety measures.
-
-[[enable-dynamic-scripting]]
-[float]
-=== Enabling dynamic scripting
-
-The `script.*` settings allow for <<security-script-fine,fine-grained>>
-control of which script languages (e.g `painless`) are allowed to
-run in which context ( e.g. `search`, `aggs`, `update`), and where the script
-source is allowed to come from (i.e. `inline`, `stored`, `file`).
-
-For instance, the following setting enables `stored` `update` scripts for
-`painless`:
-
-[source,yaml]
-----------------
-script.engine.painless.inline.update: true
-----------------
-
-Less fine-grained settings exist which allow you to enable or disable scripts
-for all sources, all languages, or all contexts.  The following settings
-enable `inline` and `stored` scripts for all languages in all contexts:
-
-[source,yaml]
------------------------------------
-script.inline: true
-script.stored: true
------------------------------------
-
-WARNING:  The above settings mean that anybody who can send requests to your
-Elasticsearch instance can run whatever scripts they choose! This is a
-security risk and may well lead to your Elasticsearch cluster being
-compromised.
+While we make every effort to prevent scrips from running amok, security is
+something you do in
+https://en.wikipedia.org/wiki/Defense_in_depth_(computing)[layers], so you'll
+want to take some steps as well.
+
+First and foremost, never run Elasticsearch as the `root` user, as this would
+allow a script to access or do *anything* on your server, without limitations.
+Elasticsearch checks if it is running as `root` on startup but it won't
+refuse to start if it can't be sure so do your part to make sure it isn't
+running as `root`.
+
+You should not expose Elasticsearch directly to users, but instead have an
+application in between your users and Elasticsearch. If you *must* allow users
```

Comment:
"an application to sanitise user input before running requests on ES" - or something similar